Base/res/js/Spreadsheet/runtime.js

```diff
@@ -365,6 +365,9 @@ function randRange(min, max) {
 }
 
 function integer(value) {
+    const typeVal = typeof value;
+    if ((typeVal !== "number" && typeVal !== "string") || Number.isNaN(value))
```

Comment:
```js
> Number.isNaN('NaN')
false
> Number.isNaN('I was told I could find NaNs here, but it\'s all numbers')
false
```

Reply:
Why would this be a problem? this should be caught by the earlier condition, the reason I had to include the second part is becuase `typeof NaN === 'number'`

Reply:
because:
```js
> function integer(value) {
    const typeVal = typeof value;
    if ((typeVal !== "number" && typeVal !== "string") || Number.isNaN(value))
        throw new Error(`integer() called with unexpected type "${typeVal}"`);
    return value | 0;
}
undefined
> integer('is this NaN yet?')
0
```

Reply:
Ohh, what do you think is a better fix in this situation?

Reply:
just a simple `Number.isNaN(Number(value))` should do I think.

Reply:
ok committed ty :)